nrf_modem: doc: PDN doc update (#411)
nrfconnect/sdk-nrfxlib: nrf_modem/doc/pdn_extension.rst

In this example, the DNS hints are extended with an extra linked-list hint that specifies the APN to use for the query.

Like for data sockets, the APN name must be set up and activated before the DNS query.

> Otherwise, the query will fallback on the default PDN if it exist.

Author: A bit unsure about this one, this is what's happens today. Is this the correct behavior or is the old documentation describing the correct behavior and we should update the implementation?

Reviewer: I would like `getaddrinfo()` to fail with `ENOTCONN` if a PDN socket does not exist for this APN, and `ENETDOWN` if the PDN socket exists but is not active.

Reviewer 2: No idea about this question, but a language fix in case the changed version is what should be used. ;)
```suggestion
Otherwise, the query will fall back on the default PDN (if it exists).
```

   at_fd = nrf_socket(NRF_AF_LTE, 0, NRF_NPROTO_AT);
   nrf_send(at_fd, "AT+CGEREP=1", strlen("AT+CGEREP=1"));
   nrf_recv(at_fd, buf, sizeof(buf), 0);

   if (strcmp("OK", buf) == 0) {

>         /* If the command did not return OK we failed to subscribe on the

Reviewer 2: ```suggestion
        /* If the command did not return OK, we failed to subscribe to the
```

         * packet domain notifications.
         */
        nrf_close(at_fd);
        return -1;
   }

   # Set up PDN socket using the APN (PDN initialize and connect)

   nrf_getsockopt(pdn_fd, NRF_SOL_PDN, NRF_SO_PDN_CONTEXT_ID, &context_id, sizeof(uint8_t));

   char pdn_notification[20];
   snprintf(pdn_notification, 20, "+CGEV IPV6 %d", context_id);

   int max_bytes;
   do {
        int recv_data = nrf_recv(at_fd, buf, sizeof(buf), 0);
        if(recv_data < 0) {
                /* nrf_recv failed for reason, this should be handled */
                return -1;
        }
        max_bytes = MIN(recv_data, sizeof(pdn_notification));
   } while(strncmp(buf, max_bytes, pdn_notification) != 0)

   nrf_close(at_fd);

When requesting an IPv6 connection (which requires PDN type IPV6 or IPV4V6), the IPv6 address might not be global even if the request is successful.
The reason for this is that many networks use SLAAC, where the prefix of the IPv6 address is set after the PDN connection is created.
Therefore, you might need to wait before using an IPv6 socket until the prefix is properly set.

> If the address is obtained, a ``+CGEV: IPV6 <cid>`` notification is received from the AT socket, if not a ``+CGEV: IPV6 FAIL <cid>`` notification is received.

Reviewer 2: ```suggestion
If the address can be obtained, the AT socket sends a ``+CGEV: IPV6 <cid>`` notification.
Otherwise, it sends a ``+CGEV: IPV6 FAIL <cid>`` notification.
```

The example code shows how to set up an AT socket that logs "CGEV" entries.
It signals when IPv6 is available for use by an IPv6 socket, either while connecting to a PDN or afterwards.

> .. note::
>         This example code does not cover the default PDN (CID 0), because this PDN is usually available in most networks.
>         However, the algorithm for checking that IPv6 is available on the default PDN is the same as if it was manually set up with an APN.

Reviewer 2: ```suggestion
.. note::
   This example code does not cover the default PDN (CID 0), because this PDN is usually available in most networks.
   However, the algorithm for checking that IPv6 is available on the default PDN is the same as if it was manually set up with an APN.
```
